cmd/xl-v1-multipart.go

```diff
@@ -491,7 +491,7 @@ func (xl xlObjects) ListObjectParts(ctx context.Context, bucket, object, uploadI
 	// Hold lock so that there is no competing
 	// abort-multipart-upload or complete-multipart-upload.
 	uploadIDLock := xl.nsMutex.NewNSLock(minioMetaMultipartBucket,
-		xl.getUploadIDDir(bucket, object, uploadID))
+		pathJoin(bucket, object, uploadID))
```

Comment:
we can define a new function `xl.getUploadIDLockPath(bucket, object, uploadID)` it can prevent unintentional programmer bugs

Reply:
Also the function can have a comment on why it is used instead of xl.getUploadIDDir for locking purposes

Reply:
done